pkg/activator/certificate/cache.go

```diff
@@ -50,68 +56,138 @@ type CertCache struct {
 
 // NewCertCache creates and starts the certificate cache that watches Activators certificate.
 func NewCertCache(ctx context.Context) (*CertCache, error) {
-	secretInformer := secretinformer.Get(ctx)
+	nsSecretInformer := nssecretinformer.Get(ctx)
+	nsConfigmapInformer := nsconfigmapinformer.Get(ctx)
 
 	cr := &CertCache{
-		secretInformer: secretInformer,
-		logger:         logging.FromContext(ctx),
+		secretInformer:    nsSecretInformer,
+		configmapInformer: nsConfigmapInformer,
```

Comment:
Curious - I'm wondering if it's simpler to just watch (or poll) a certain directory for certs instead of using informers here. 

Then in theory we wouldn't necessarily be tied to 'trust-manager` and could swap to use k8s cluster trust bundles ([ref](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3257-cluster-trust-bundles)).

It's going beta in 1.30

Reply:
We could, we do it in Queue-Proxy already. Having multiple config-maps is not tying only to trust-manager.We also have downstream solutions with the same pattern (https://github.com/ReToCode/knative-encryption/tree/main/8-trust-sources), so we'll have multiple ConfigMaps to look at.
If we do it via filesystem, we'd need a way to configure  multiple mount-configs somehow. But I'm up for looking at this again once cluster trust bundles landed (we'd also need support for this downstream).

Reply:
> If we do it via filesystem, we'd need a way to configure multiple mount-configs somehow. But I'm up for looking at this again once cluster trust bundles landed (we'd also need support for this downstream).

I think it might be simple enough for operators to add a projected volume mount ([ref](https://kubernetes.io/docs/concepts/storage/projected-volumes/)) coalescing all the CA's into the same volume mount. If we use the standard `SSL_CERT_DIR` then I think the activator might simply need to call `x509.SystemCertPool()` every few minutes (https://pkg.go.dev/crypto/x509#SystemCertPool) to get updates.

One thing to verify is if projected volumes receive updates when config maps change - if they don't we should file a k8s bug.

Reply:
> I think it might be simple enough for operators to add a projected volume mount ([ref](https://kubernetes.io/docs/concepts/storage/projected-volumes/)) coalescing all the CA's into the same volume mount.

I think that would work. But how would a user define the volume-mounts? He'd need to customize our yamls on installation (and on every update) to add the volume mounts. Also operator would probably need some form of extra-mounts attributes for all containers. Isn't that more complicated that what we have in this PR?

Reply:
We require operators do this for tag-to-digest resolution - https://knative.dev/docs/serving/tag-resolution/#custom-certificates

Reply:
I think for now I'd rather push this complexity to users because long term I think advanced operators would probably need to configure other activator options manually (https://github.com/knative/serving/issues/14633, https://github.com/knative/serving/issues/14634)

Reply:
In this PR we also trust the [CA in a Secret](https://github.com/knative/serving/pull/14717/files#diff-d4fc9e56478eac3333afe9c0ea167afa2d38ba5d74454a79712400c571ef2cf7R144) (if present), I think it's a good thing to make the "easy" setup work without additional configuration. That certificate will not be in the system pool by default, so users always would need to give Activator the CA in that case. So I'm not (yet) really convinced that is good UX for our users.

Reply:
The CA is not always populated, also what component would manage that volume in the YAML? The secret only exists when encryption is enabled and cert-manager is deployed. I'd assume the deployment would fail if no Secret exists. Even when it exists, the CA can be empty depending on the cert-manager issuer type (only CA and self-signed populate the `ca.crt` field). So we'd mount an empty file.

Reply:
I did think about this a bit more and have another reason to keep this PR. This affects not just serving but also the net-* implementations. With the current solution, a user just provides the CM (ideally also using trust-manager to distribute it in multiple namespaces) and it works directly. No changes necessary to the YAML (that one has to do on every upgrade), no further changes on the operator and the KnativeServing CR.

With the mount approach, a user would need to customize the volumes of controller and net-* controller for now. We might at some point add mTLS, then that would also be needed on every KnativeService in the QP manifest. So I'm not sure this is really a good alternative as it has way worse UX.